### PR TITLE
 ISSUE-883 continue cluster management with namespace

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Support cluster management with namespace for databend query

## Changelog

- New Feature

## Related Issues
part #883

### CODE REVIEW

File: query/src/clusters/cluster.rs [hunk 1]
```diff
@@ -12,140 +12,233 @@
...
-    pub fn create_global(cfg: Config) -> Result<ClusterRef> {
-        Ok(Arc::new(Cluster {
-            nodes: Mutex::new(HashMap::new()),
-            local_port: Address::create(&cfg.query.flight_api_address)?.port(),
+impl ClusterDiscovery {
+    // TODO(Winter): this should be disabled by compile flag
+    async fn standalone_without_metastore(cfg: &Config) -> Result<ClusterDiscoveryRef> {
+        let local_id = global_unique_id();
+
+        let local_store = LocalKVStore::new_temp().await?;
```
Comment (BohuTANG) [inline query/src/clusters/cluster.rs:47]: This api will change from `async` to `sync`.
I think all the meta api all `sync` is ok.
cc @dantengsky @zhang2014

Comment (zhang2014) [reply query/src/clusters/cluster.rs]: got it. by the way I think `async` is OK if it is not `&mut self`

File: common/management/src/namespace/namespace_mgr.rs [hunk 2]
```diff
@@ -32,109 +34,208 @@ pub static NAMESPACE_API_KEY_PREFIX: &str = "__fd_namespaces";
...
+        kv_api: Arc<dyn KVApi>,
+        tenant: &str,
+        namespace: &str,
+        lift_time: Duration,
+    ) -> Result<Self> {
+        Ok(NamespaceMgr {
+            kv_api,
+            lift_time,
+            namespace_prefix: format!(
+                "{}/{}/{}/databend_query",
```
Comment (BohuTANG) [inline common/management/src/namespace/namespace_mgr.rs:52]: For the prefix: `"{}/{}/{}/databend_query"`,  the suffix `databend_query` is used for?

**Pull request merged**